src/Mentat.Domain/Bash/BashTestConfig.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Mentat.Domain.Interfaces;
+
+namespace Mentat.Domain.Bash
+{
+    public class BashTestConfig : IBashTestConfig
+    {
+        public int NumberOfTests { get; private set; }
+
+        public string Language { get; private set; }
+
+        public IEnumerable<string> TestFileNames { get; private set; }
+
+        public string SampleExecutableName { get; private set; }
+
+        public float TimeoutInSeconds { get; private set; }
+
+        public string DiffCommand { get; private set; }
+
+        public BashTestConfig()
+        {
+            Language = string.Empty;
+            TestFileNames = new List<string>();
+            NumberOfTests = TestFileNames.Count();
```

Comment:
after changing above comment, remove this plus all other occurences of directly assigning numberOfTests.